### PR TITLE
Added last read/wrote timestamp metrics to loki.write

### PR DESCRIPTION
**Pull request by thepalbi**
#### PR Description

This PR adds two new metrics to the WAL enabled `loki.write`:
- last written timestamp
- last read timestamp

This will allow us to track the delay between the WAL writing and reading side, in actual time from the entries flowing through the WAL.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: component/common/loki/client/queue_client.go [hunk 1]
```diff
@@ -283,16 +285,26 @@ func (c *queueClient) AppendEntries(entries wal.RefEntries, segment int) error {
...
 		c.markerHandler.UpdateReceivedData(segment, len(entries.Entries))
 	} else {
 		// TODO(thepalbi): Add metric here
 		level.Debug(c.logger).Log("msg", "series for entry not found")
 	}
+
+	// emit metric tracking max seen timestamp in WAL entry
+	if maxSeenTimestamp != -1 {
+		c.qcMetrics.lastReadTimestamp.WithLabelValues().Set(float64(maxSeenTimestamp))
+	}
```
Comment (tpaschalis) [inline component/common/loki/client/queue_client.go:306]: nit: won't this _always_ be the case when there's at least 1 entry?

File: pkg/util/metrics.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package util
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// MustRegisterOrGet will attempt to register the supplied collector into the register. If it's already
+// registered, it will return that one.
+// In case that the register procedure fails with something other than already registered, this will panic.
+func MustRegisterOrGet(reg prometheus.Registerer, c prometheus.Collector) prometheus.Collector {
```
Comment (tpaschalis) [inline pkg/util/metrics.go:8]: Good call; we can reuse this in other places inside of the Loki components

**Review by tpaschalis — APPROVED** [review]
LGTM with a small nit

File: component/common/loki/client/queue_client.go (hunk 1 above)
Comment (thepalbi) [reply component/common/loki/client/queue_client.go]: Yes! Upon an `AppendEntries` call there will always be at least one entry, if not it wouldn't have been decoded. Removing if.

**Comment by thepalbi** [comment]
@tpaschalis all fixed

**Pull request merged**